pyenv-win/bin/pyenv.bat

```diff
@@ -1,16 +1,103 @@
 @echo off
-setlocal enableDelayedExpansion
+setlocal
 chcp 1250 >nul
 
-IF EXIST "%~dp0"..\exec.bat (
-   del /F /Q "%~dp0"..\exec.bat >nul
+set "pyenv=cscript //nologo "%~dp0..\libexec\pyenv.vbs""
+
+rem if 'pyenv' called alone, then run pyenv.vbs
+if [%1]==[] (
+  %pyenv%
+  exit /b
 )
 
-call cscript //nologo "%~dp0"..\libexec\pyenv.vbs %*
-IF EXIST "%~dp0"..\exec.bat (
-   call "%~dp0"..\exec.bat
+rem use pyenv.vbs to aid resolving absolute path of "active" version into 'bindir'
+set "bindir="
+for /f %%i in ('%pyenv% version') do call :normalizepath "%~dp0..\versions\%%i" bindir
+
+rem all help implemented as plugin
+if /i [%2]==[--help] goto :plugin
+
+rem let pyenv.vbs handle these
+set commands=rehash global local version vname version-name versions commands shims which whence help --help
+for %%a in (%commands%) do (
+  if /i [%1]==[%%a] (
+    %pyenv% %*
+    exit /b
+  )
 )
-for /f "tokens=1 delims=|" %%A in (""!PYENV_VERSION!"") do (
-   endLocal
-   set "PYENV_VERSION=%%~A"
+
+rem jump to plugin or fall to exec
+if /i not [%1]==[exec] goto :plugin
+
+
+rem ====================================================================================
+:exec
+
+echo %bindir%
```

Comment:
Should be removed, no?

Reply:
Yes it should, and also script should exit when no active version. Will fix shortly.